redirector/common.py

```diff
@@ -80,13 +80,26 @@ def get_ddb_table():
     return boto3.resource("dynamodb").Table(MANIFEST_TABLE_NAME)
 
 
+def get_file_pathname(file):
+    # want to start the path after "All Files/<BoxFolderName>/"
+    file_path_collection = file.path_collection
+    file_path_names = [fp.get().name for fp in file_path_collection["entries"][2:]]
+    if len(file_path_names) > 0:
+        file_path = "/".join(file_path_names)
+        fullpath = "/".join([file_path, file.name])
+    else:
+        fullpath = file.name
+
+    return fullpath
+
+
 def put_file_item(ddb_table, file):
     assert is_box_file_public(
         file
     ), "cannot put a file that hasn't been shared publicly"
 
     item = {
-        "filename": file.name,
+        "filename": get_file_pathname(file),
```

Comment:
Maybe we should change the DynamoDB key name to "path" or the like, so it's clear what we're storing there?

Reply:
dc7f74c and c2785eb should cover this change.

Reply:
also 9ed350f